### PR TITLE
Port Spring Web to RESTEasy Reactive

### PR DESCRIPTION
**Pull request by geoand**
This means that the Spring Web can now longer
be used with RESTEasy Classic which means
the loss of the following functionality:

* XML support via the JAX-B extension
* Support for Servlet types

The port however brings the following benefits:

* Aligns with the move to make RESTEasy Reactive the default REST layer
* Improves runtime performance
* Allows up to better support this module as all the necessary code is owned by Quarkus instead of being split between Quarkus and RESTEasy Classic

### CODE REVIEW

**Comment by geoand** [comment]
@glefloch the gradle test failures seem unrelated, no?

**Comment by glefloch** [comment]
@geoand yes, they are not related to this PR.

**Comment by geoand** [comment]
Thanks for confirmation!

File: independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java [hunk 1]
```diff
@@ -37,17 +37,27 @@
 public class ResteasyReactiveScanner {
 
     public static final Map<DotName, String> BUILTIN_HTTP_ANNOTATIONS_TO_METHOD;
+    public static final Map<String, DotName> METHOD_TO_BUILTIN_HTTP_ANNOTATIONS;
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java:40]: Where is this used?

Comment (gastaldi) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java:40]: Nevermind, found it in https://github.com/quarkusio/quarkus/pull/20195/files#diff-4e34a8d1bf348000ad6478d0e6152cc5728dd01bce175e3ed38461f80eba1d8cR182

**Review by gastaldi — APPROVED** [review]
Looks good! Awesome job! 🥳

**Comment by geoand** [comment]
@stuartwdouglas any objections to this?

**Review by stuartwdouglas — APPROVED** [review]
LGTM, but I don't really know anything about Spring.

**Comment by geoand** [comment]
Sure, I just wanted to know whether you had any general objections :)

**Pull request merged**